pkg/gateway/servicemap.go

```diff
@@ -41,3 +43,24 @@ type Endpoint struct {
 	IP   string
 	Port int
 }
+
+type Aliases struct {
+	Data map[string]string
+}
+
+// Generates a list of aliases for a given ingress name and namespace.
+func (a *Aliases) Collect(name string, ns string) []string {
+	results := make([]string, 0)
+	for key, value := range a.Data {
+		if value == fmt.Sprintf("%s.%s", name, ns) {
+			results = append(results, key)
+		}
+	}
+	return results
+}
+
+// Generates a concatenated list of aliases for a given ingress name and namespace.
+func (a *Aliases) AliasNames(name string, ns string) string {
```

Comment:
This doesn't seem super valuable to have attached to the `Aliases` type. It's really specific to the way we need to format names for the nginx template. I would suggest we either run this Join operation within the template itself, or we do it in the code that calls Execute on the template.